plugin/core/settings.py

```diff
@@ -157,53 +157,78 @@ def unload_settings() -> None:
         _settings_obj.clear_on_change("_on_new_client_settings")
 
 
-def read_language_config(config: dict) -> LanguageConfig:
-    language_id = config.get("languageId", "")
-    scopes = config.get("scopes", [])
-    syntaxes = config.get("syntaxes", [])
-    return LanguageConfig(language_id, scopes, syntaxes)
+def convert_syntaxes_to_selector(d: Dict[str, Any]) -> Optional[str]:
```

Comment:
Because there's two ways to specify languages in JSON. One way is to put languageId/syntaxes/document_selector at the same level as command/enabled/env. Other way is to have languages array.